openai-model-provider/proxy/proxy.go

```
@@ -0,0 +1,97 @@
+package proxy
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+)
+
+type Config struct {
+	APIKey          string
+	Port            string
+	UpstreamHost    string
+	UseTLS          bool
+	ValidateFn      func(cfg *Config) error
+	RewriteModelsFn func(*http.Response) error
+	PathPrefix      string
+}
+
+type server struct {
+	cfg *Config
+}
+
+func Run(cfg *Config) error {
+	if cfg.Port == "" {
+		cfg.Port = "8000"
+	}
+	if cfg.UpstreamHost == "" {
+		cfg.UpstreamHost = "api.openai.com"
+		cfg.UseTLS = true
+	}
+
+	if cfg.RewriteModelsFn == nil {
+		cfg.RewriteModelsFn = DefaultRewriteModelsResponse
+	}
+
+	if cfg.ValidateFn != nil {
+		if err := cfg.ValidateFn(cfg); err != nil {
+			return fmt.Errorf("validation failed: %w", err)
+		}
```

If I saw that correctly, all validation functions log the error but return `nil`, so this will never trigger (but likely should at this point?)

I think you can just return the errors directly in the validation functions and only in your validation func handle the error:

```go
func Validate(cfg *Config) error {
	if cfg.ValidateFn == nil {
		return nil
	}
	return handleValidationError(cfg.ValidateFn(cfg)) // this will be the only point calling handleValidationError... also, you can still log the error here if you want to (just need to figure out a way to get the loggerpath here or just include it in the err)
}
```

e.g. in the vLLM validation func you would then do 

```go
if cfg.APIKey == "" {
		return fmt.Errorf("missing vLLM APIKey") // text only needed if you're going to bubble it up
	}
```